question/bank/managecategories/classes/question_category_object.php

```diff
@@ -0,0 +1,518 @@
+     */
+    public $editlists = array();
+
+    /**
+     * @var string tab.
+     */
+    public $tab;
+
+    /**
+     * @var int tab size.
+     */
+    public $tabsize = 3;
+
+    /**
+     * @var moodle_url Object representing url for this page
+     */
+    public $pageurl;
+
+    /**
+     * @var question_category_edit_form Object representing form for adding / editing categories.
+     */
+    public $catform;
+
+    /**
+     * Constructor.
+     *
+     * @param int $page page number.
+     * @param moodle_url $pageurl base URL of the display categories page. Used for redirects.
+     * @param context[] $contexts contexts where the current user can edit categories.
+     * @param int $currentcat id of the category to be edited. 0 if none.
+     * @param int|null $defaultcategory id of the current category. null if none.
+     * @param int $todelete id of the category to delete. 0 if none.
+     * @param context[] $addcontexts contexts where the current user can add questions.
+     */
+    public function __construct($page, $pageurl, $contexts, $currentcat, $defaultcategory, $todelete, $addcontexts) {
+
+        $this->tab = str_repeat('&nbsp;', $this->tabsize);
+
+        $this->str = new stdClass();
+        $this->str->course         = get_string('course');
+        $this->str->category       = get_string('category', 'question');
+        $this->str->categoryinfo   = get_string('categoryinfo', 'question');
+        $this->str->questions      = get_string('questions', 'question');
+        $this->str->add            = get_string('add');
+        $this->str->delete         = get_string('delete');
+        $this->str->moveup         = get_string('moveup');
+        $this->str->movedown       = get_string('movedown');
+        $this->str->edit           = get_string('editthiscategory', 'question');
+        $this->str->hide           = get_string('hide');
+        $this->str->order          = get_string('order');
+        $this->str->parent         = get_string('parent', 'question');
+        $this->str->add            = get_string('add');
+        $this->str->action         = get_string('action');
+        $this->str->top            = get_string('top');
+        $this->str->addcategory    = get_string('addcategory', 'question');
+        $this->str->editcategory   = get_string('editcategory', 'question');
+        $this->str->cancel         = get_string('cancel');
+        $this->str->editcategories = get_string('editcategories', 'question');
+        $this->str->page           = get_string('page');
+
+        $this->pageurl = $pageurl;
+
+        $this->initialize($page, $contexts, $currentcat, $defaultcategory, $todelete, $addcontexts);
+    }
+
+    /**
+     * Old syntax of class constructor. Deprecated in PHP7.
+     *
+     * @deprecated since Moodle 3.1
+     */
+    public function question_category_object($page, $pageurl, $contexts, $currentcat, $defaultcategory, $todelete, $addcontexts) {
```

Comment:
do we need this deprecated one here? its already there in the old one and any legacy code is not pointing to this code.